Lib/test/test_list.py

```diff
@@ -196,6 +197,63 @@ def test_preallocation(self):
         self.assertEqual(iter_size, sys.getsizeof(list([0] * 10)))
         self.assertEqual(iter_size, sys.getsizeof(list(range(10))))
 
+    @cpython_only
+    def test_overallocation(self):
+        # bpo-33234: Don't overallocate when initialized from known lengths
+        # bpo-38373: Allows list over-allocation to be zero for some lengths
+        # bpo-43574: Don't overallocate for list-literals
+        sizeof = sys.getsizeof
+
+        # First handle empty list and empty list-literal cases.  Should have no
+        # overallocation, including init from iterable of unknown length.
+        self.assertEqual(sizeof([]), sizeof(list()))
+        self.assertEqual(sizeof([]), sizeof(list(tuple())))
+        self.assertEqual(sizeof([]), sizeof(list(x for x in [])))
+
+        # Must use actual list-literals to test the overallocation behavior of
+        # compiled list-literals as well as those initialized from them.
+        test_literals = [
+                [1],
+                [1,2],
+                [1,2,3],  # Literals of length > 2 are special-cased in compile
+                [1,2,3,4],
+                [1,2,3,4,5,6,7],
+                [1,2,3,4,5,6,7,8], # bpo-38373: Length 8 init won't over-alloc
+                [1,2,3,4,5,6,7,8,9],
+                ]
+
+        overalloc_amts = []
+        for literal in test_literals:
```

Comment:
no action required: meta: this is where I really wish we had an `absltest.paramtereize` or `pytest.mark.parameterize` available in the stdlib.  it'd run all of these as unique tests with appropriate names instead of a loop that prevents others from running upon first failure.

Reply:
I think unittest.subTest() displays parameters on the failing subtests and keeps running through each even on failure.